tracer/src/Datadog.Trace/Telemetry/Collectors/MetricsTelemetryCollector.cs

```diff
@@ -110,6 +110,13 @@ public MetricResults GetMetrics()
         return new(metricData, distributionData);
     }
 
+    public void Clear()
+    {
+        _reserveBuffer.Clear();
+        var buffer = Interlocked.Exchange(ref _buffer, _reserveBuffer);
```

Comment:
I am not sure if I understand the reason for the `Interlocked.Exchange` over just doing a `_buffer.Clear()`?

Reply:
`Clear()` isn't thread-safe, so it shouldn't be called on an "active" buffer. This attempts to ensure we don't try to clear at the same time as something is recording to the active buffer.

Although, it raises the fact there's still an edge case here where we _could_ call `Clear()` at the same time as recording a metric if we're very unlucky🤔 It _shouldn't_ be an issue in practice, but maybe `Clear()` is too risky anyway, will have a think, thanks!

Reply:
Ahh okay that helps clear it up thanks!

Reply:
> maybe Clear() is too risky anyway, will have a think

Fixed this in `https://github.com/DataDog/dd-trace-dotnet/pull/4198/commits/b99a481f71aded3f440c549f15665b7b232f996b` by calling `TryDequeue` in a loop instead of `Clear()`